### ISSUE TITLE
Builder pattern for configs

### ISSUE THREAD

**Issue by Shourya742** [citadel-tech/coinswap#235]
Currently, our configs for both the taker and maker are quite verbose. We don’t want all the config fields to be accessible by third parties; only a specific set of fields should be available. Since the new method for initializing these structs is quite lengthy, we plan to use the builder design pattern. This will make it easier for third parties to modify the configurations with minimal effort. Change the config for both, to follow this.

**Comment by DaCatBeam** [comment]
@Shourya742 Are you looking to retain the default behavior of writing config files for the Maker and Taker Builders on non-existence of the config files?

**Comment by Shourya742** [comment]
@DaCatBeam We gonna still retain the default behaviour, but gonna make the initialization a bit more convenient.

**Comment by KnowWhoami** [comment]
@Shourya742 
Do we require Builder pattern at this stage??
I think , it's not required at least for v0.1.2 since we don't have much  users who are integrating our coinswap library into their downstream products..

**Comment by KnowWhoami** [comment]
Yeah, I think it would increase the interface of handling config related stuffs in the codebase at least and might simplify some the api's ..

**Comment by KnowWhoami** [comment]
Agree that ,  this would involve creating better design and would require deeper thought on it..
what do you think @mojoX911 ??

**Comment by mojoX911** [comment]
Ack, on being good to have but not high priority right now as we do not have any downstream library users. Whether we will get them or how they will interact with the lib is also an open question, so the builder pattern seems premature optimization at this stage.

We can tackle it once the need arises.

**Comment by mojoX911** [comment]
Won't be in priority for the next few releases. Our configs are already quite small. Builder pattern is overkill at this stage. If need arises we will revisit this again.
